lib/logthrdest/logthrdestdrv.c

```diff
@@ -1058,12 +1052,23 @@ _create_workers(LogThreadedDestDriver *self)
   return TRUE;
 }
 
+static void
+_assert_on_persist_name(LogThreadedDestDriver *self)
+{
+  // Make sure that generate_persist_name virtual function still returns
+  // the same value as provided during the init phase. Because several
+  // functionality of the LogThreadedDestDriver depends on it.
+  g_assert_cmpstr(self->persist_name_for_assertion, ==, log_pipe_get_persist_name((LogPipe *)self));
+}
+
 gboolean
-log_threaded_dest_driver_init_method(LogPipe *s)
+log_threaded_dest_driver_init_method(LogPipe *s, const gchar *persist_name)
```

Comment:
honestly, I don't really like that we are changing the signature of this function, this is just the implementation of the log_pipe_init() interface, which a derived function needs to call, so in a way, it is the equivalent of calling the super()'s same-named method in Python. I can see that you have added a wrapper to make this compatible with log_pipe_init()

I understand that you want to be explicit about the dependency, but if that's the case, then it might make sense to make log_pipe_init of LogThreadedDestDriver non-overridable (as if you declared it private in C++) and create a new layer for the initialization of threaded dest drivers, which derived classes can hook into.

How that new layer (method) could be named, I have no clue at this point, but then it's only a naming issue :)